modules/l4fail2ban/matcher.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2024 SICK AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package l4fail2ban
+
+import (
+	"fmt"
+	"net"
+	"net/netip"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+
+	caddy_fail2ban "github.com/Javex/caddy-fail2ban"
+	"github.com/mholt/caddy-l4/layer4"
+	"go.uber.org/zap"
+)
+
+func init() {
+	caddy.RegisterModule(&Fail2Ban{})
+}
+
+type Fail2Ban struct {
+	Banfile string `json:"banfile"`
+
+	logger  *zap.Logger
+	banlist caddy_fail2ban.Banlist
+}
+
+// CaddyModule returns the Caddy module information.
+func (*Fail2Ban) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID:  "layer4.matchers.fail2ban",
+		New: func() caddy.Module { return new(Fail2Ban) },
+	}
+}
+
+// Provision implements caddy.Provisioner.
+func (m *Fail2Ban) Provision(ctx caddy.Context) error {
+	m.logger = ctx.Logger()
+
+	// Create new banlist, same as in http.matchers.fail2ban (https://github.com/Javex/caddy-fail2ban/blob/main/banlist.go)
+	m.banlist = caddy_fail2ban.NewBanlist(ctx, m.logger, &m.Banfile)
+	m.banlist.Start()
+	return nil
+}
+
+// The Match will return true if the remote IP is found in the ban list
+func (m *Fail2Ban) Match(cx *layer4.Connection) (bool, error) {
+	clientIP, err := m.getRemoteIP(cx)
+	strClientIP := clientIP.String()
```

Comment:
I suppose there is no point in converting clientIP into string if err != nil. So I would move this line down, just before the IsBanned() condition.

Reply:
Good catch, I just moved the line down.